src/maybe.js

```diff
@@ -0,0 +1,31 @@
+'lang sweet.js';
+// @flow
+
+export class Maybe<T> {
+  static of(v: T): Maybe<T> {
+    return new Just(v);
+  }
+
+  map<V>(f: T => V): Maybe<V> {
+    throw new Error();
+  }
+}
```

Comment:
We may want to implement `Monoid`, `Monad` and `Alternative` for `Maybe` as well. Give the user more options.